analyses/deep-review-contrib/README.md

```diff
@@ -9,3 +9,5 @@ conda env create --file environment.yml
 source activate contrib-viz
 Rscript install-manual-r-dependencies.r
 ```
+
+The environment was updated to be compatible with Binder after the notebooks were originally run.
```

Comment:
Do you want to rerun the notebooks and save with the current environment?

Reply:
This was intended as a temporary comment that we can remove after rerunning the notebooks in the new environment.  However, I'm out of the office today and may not trust that the environment works perfectly on this Windows laptop.

I also thought that we might add a dot plot for https://github.com/greenelab/meta-review/issues/149 and then run the notebook one final time after that update.  What do you think?

Reply:
> I also thought that we might add a dot plot for #149 and then run the notebook one final time after that update. What do you think?

Yesterday I quickly added `geom_dot()` in place in the figure code, but the dots didn't align correctly with the filled region. `geom_dot()` without the lines worked, but the combination was wonky. I don't have an example to show because I was just messing around in Binder. Perhaps @dhimmel can easily figure out what's up.

Reply:
I'm quite bad with R plotting so I'm not planning to make an attempt at this.

Reply:
FYI, I just tried to launch Binder using the link at the top of this page, to recreate the figure that I failed to describe with words and Binder failed.

![image](https://user-images.githubusercontent.com/5393995/53830056-75e16380-3f36-11e9-8e49-8999777cb754.png)

Reply:
End of logs show...

```
* DONE (ggridges)
Removing intermediate container e636367f983b
 ---> 6d68f48f80dd
Step 47/47 : CMD ["jupyter", "notebook", "--ip", "0.0.0.0"]
 ---> Running in 5785ed11313d
Removing intermediate container 5785ed11313d
 ---> 1bd6c22dcbe3
{"aux": {"ID": "sha256:1bd6c22dcbe378bf58007f2cab2085ccbaf6006e7963fd7726f285c1ae2667ab"}}Successfully built 1bd6c22dcbe3
Successfully tagged gcr.io/binder-prod/r2d-72d7634-agitter-2dmeta-2dreview-39aa7f:7512bcd452a0cbb6dd6241a296b69be9f51475ac
Pushing image
Pushing image
Pushing image
Pushing image
Pushing image
Pushing image
Pushing image
Pushing image
Pushing image
Pushing image
Pushing image
Failed to connect to event stream
```

Reply:
> I also thought that we might add a dot plot for #149 and then run the notebook one final time after that update. What do you think?

I'm happy to take a crack at this visualization. I suggest we merge this PR first so I can work off of the updated environment.

Reply:
```r
gg = total_words_df %>%
  ggplot2::ggplot(ggplot2::aes(x = date, y = username, height = total_words, group = username, fill=username, color=username)) +
  ggplot2::geom_dotplot(binaxis = "y", dotsize = 0.1) + 
  # ggridges::geom_density_ridges(stat='identity', size=0.4, alpha=0.3) +
  ggplot2::scale_x_datetime(expand = c(0, 0), date_labels = '%b %Y', breaks = major_breaks, minor_breaks = minor_breaks) +
  ggplot2::scale_y_discrete(limits=usernames, position = 'right', labels=y_labels) +
  #ggplot2::scale_fill_manual(values = setNames(usage_df$color, usage_df$username)) +
  ggplot2::guides(fill=FALSE, color=FALSE) +
  ggplot2::xlab(NULL) +
  ggplot2::ylab(NULL) +
  ggplot2::theme_minimal() +
  ggplot2::theme(axis.text.y = ggplot2::element_text(vjust = 0)) +
  ggplot2::theme(panel.grid.major.y = ggplot2::element_blank())
```

This naively looks correct, but dot plot starts around Jul 2017 instead of Jan 2017.

![image](https://user-images.githubusercontent.com/5393995/53832723-c4ddc780-3f3b-11e9-8b94-362b419e0904.png)

Binning in the other direction looks correct, but it's not really the "timeline" view that the reviewer requested.

```r
gg = total_words_df %>%
  ggplot2::ggplot(ggplot2::aes(x = date, y = username, height = total_words, group = username, fill=username, color=username)) +
  ggplot2::geom_dotplot(binaxis = "x", dotsize = 0.2) + 
  ggridges::geom_density_ridges(stat='identity', size=0.4, alpha=0.3) +
  ggplot2::scale_x_datetime(expand = c(0, 0), date_labels = '%b %Y', breaks = major_breaks, minor_breaks = minor_breaks) +
  ggplot2::scale_y_discrete(limits=usernames, position = 'right', labels=y_labels) +
  ggplot2::guides(fill=FALSE, color=FALSE) +
  ggplot2::xlab(NULL) +
  ggplot2::ylab(NULL) +
  ggplot2::theme_minimal() +
  ggplot2::theme(axis.text.y = ggplot2::element_text(vjust = 0)) +
  ggplot2::theme(panel.grid.major.y = ggplot2::element_blank())
```

![image](https://user-images.githubusercontent.com/5393995/53833165-c0fe7500-3f3c-11e9-9b03-7a5c2e2ef726.png)

Reply:
@slochower `Failed to connect to event stream` is not a fatal error in my recent experience.  It indicates an interruption between the build process and the local client that prevents the log from updating in the browser, but the build continues.  Usually refreshing will either reset the build log output (from its current point, not the beginning) or launch the container if it finished building.

Reply:
Huh. A refresh launched a rebuild, which did work, but it took another 5-10 minutes, so I don't think it simply reset the output. Good to know, though!